state-and-event-handling/updating-state.md

```diff
@@ -0,0 +1,442 @@
+# Updating State
+
+## Goals
+
+As a continuation of managing state in React components, the goal of this lesson is to learn how to update state. This lesson will also apply our knowledge about handling events in React.
+
+## Updating State
+
+Let's assume that we have a component that has initialized a piece of state named `pieceOfState`:
+
+<!-- prettier-ignore-start -->
+```js
+const [pieceOfState, setPieceOfState] = useState('Initial value for pieceOfState.');
+```
+<!-- prettier-ignore-end -->
+
+In order to update state, we call the update function, and pass in the new value.
+
+<!-- prettier-ignore-start -->
+```js
+setPieceOfState('New value of pieceOfState.');
+```
+<!-- prettier-ignore-end -->
+
+| <div style="min-width:200px;"> Piece of Code </div> | Notes                                                                |
+| --------------------------------------------------- | -------------------------------------------------------------------- |
+| `setPieceOfState(...)`                              | **Replace this** with the function that was created with `useState`. |
+| `'New value of pieceOfState.'`                      | **Replace this** with the new value for the piece of state.          |
+
+This code will update a piece of state named `pieceOfState`, and its new value will be `'New value of pieceOfState.'`.
+
+### Updating State Re-Renders the Component
+
+Recall that pieces of state are responsible for managing how one piece of the component looks. For example, they can control if an element shows or hides or the classes that an element has.
+
+**Every time a piece of state is updated, the component re-renders**.
+
+### !callout-warning
+
+## Working With State and Re-Rendering
+
+Re-rendering a component is hard work! Firstly, setting state is _asynchronous_. Secondly, to repeat, _every time that state updates, the component re-renders_. These two facts will aid us when debugging state.
+
+### !end-callout
+
+## Updating State As Part of Event Handling
+
+The best time to update state is in response to _handling user events_.
+
+We can use our update functions inside an event handler.
+
+### Example: Increasing Like Count
+
+Let's imagine a `Post` component. It's responsible for displaying the number of likes that a post has.
+
+Each post starts with zero likes. When we click the "like" button, the number of likes increases.
+
+Let's consider the code we need to write for this:
+
+1. The `Post` component should have a piece of state, `likesCount`.`likesCount` is ideally a piece of state, and not a `prop`, because state can be modified within a component.
+1. We should also get a function to update this piece of state, `setLikesCount`.
```

Comment:
say what likesCount should be (changeable data) and how that implies it should be a piece of state
spacing
still gonna push for reference to function
```suggestion
1. The `Post` component should have a piece of state, `likesCount`. `likesCount` is data that changes within the component, so we should represent it with a piece of state rather than a `prop`.
1. We should get a reference to the update function for this piece of state, and store it in a variable with a good name like `setLikesCount`.
```

Reply:
Keep pushing for reference to function!! I'll try to do a sweep for that in the latest PR